From c6319fb59e0cd5b6d3d4e3401abd9c2057059a74 Mon Sep 17 00:00:00 2001
From: Jakub Luzny <jakub@luzny.cz>
Date: Tue, 11 Aug 2020 10:40:18 +0200
Subject: [PATCH] dt/dtoverlays: Add jailhouse-memory DT overlay

It is meant to reserve memory that can be used by the Jailhouse partitioning
hypervisor. More regions can be reserved.

Upstream-Status: Inappropriate [Jailhouse specific]

Signed-off-by: Jakub Luzny <jakub@luzny.cz>
---
 arch/arm/boot/dts/overlays/Makefile                |  1 +
 arch/arm/boot/dts/overlays/README                  |  7 +++++
 .../boot/dts/overlays/jailhouse-memory-overlay.dts | 30 ++++++++++++++++++++++
 3 files changed, 38 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/jailhouse-memory-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index fa6db6eed3e2..38c61c67606d 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -92,6 +92,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	iqaudio-dacplus.dtbo \
 	iqaudio-digi-wm8804-audio.dtbo \
 	irs1125.dtbo \
+	jailhouse-memory.dtbo \
 	jedec-spi-nor.dtbo \
 	justboom-both.dtbo \
 	justboom-dac.dtbo \
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 69892d01faae..4a62f204b4f9 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1528,6 +1528,13 @@ Load:   dtoverlay=irs1125
 Params: <None>
 
 
+Name:   jailhouse-memory
+Info:   Reserves memory for use by the Jailhouse hypervisor
+Load:   dtoverlay=jailhouse-memory,<param>=<val>
+Params: start                   Start of the reserved memory area (default 0x20000000)
+        size                    Size of the reserved memory area (default 0x10000000)
+
+
 Name:   jedec-spi-nor
 Info:   Adds support for JEDEC-compliant SPI NOR flash devices.  (Note: The
         "jedec,spi-nor" kernel driver was formerly known as "m25p80".)
diff --git a/arch/arm/boot/dts/overlays/jailhouse-memory-overlay.dts b/arch/arm/boot/dts/overlays/jailhouse-memory-overlay.dts
new file mode 100644
index 000000000000..26c3ce6b9e3d
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/jailhouse-memory-overlay.dts
@@ -0,0 +1,30 @@
+// Reserves memory for use by the Jailhouse hypervisor
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2835";
+
+	fragment@0 {
+		target-path = "/";
+		__overlay__ {
+
+			reserved-memory {
+				#address-cells = <2>;
+				#size-cells = <1>;
+
+				ranges;
+
+				jailhouse_memory: jailhouse@20000000 {
+					reg = <0x0 0x20000000 0x10000000>;
+					no-map;
+				};
+			};
+		};
+	};
+
+	__overrides__ {
+        start = <&jailhouse_memory>,"reg#0";
+		size  = <&jailhouse_memory>,"reg:8";
+    };
+};
-- 
2.11.0

